### PR TITLE
New arch/update public share for file

### PR DESCRIPTION
**Pull request by davigonz**
Implements https://github.com/owncloud/android/issues/2516
Needs https://github.com/owncloud/android/pull/2522

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.2'
+        classpath 'com.android.tools.build:gradle:3.4.0'
```
Comment (jcmontesmartos) [inline build.gradle]: brave!! xD

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.kt [hunk 2]
```diff
@@ -235,74 +235,74 @@ class ShareXMLParser {
...
+                remoteShare.name = readNode(parser, NODE_NAME)
 
             } else {
                 skip(parser)
             }
         }
 
-        if (isValidShare(share)) {
-            shares.add(share)
+        if (isValidShare(remoteShare)) {
```
Comment (jcmontesmartos) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.kt]: Why don't you move this function to RemoteShare class?

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt [hunk 3]
```diff
@@ -151,21 +154,29 @@ class UpdateRemoteShareOperation
...
+            val parser = ShareToRemoteOperationResultParser(
+                ShareXMLParser()
+            )
+            parser.ownCloudVersion = client.ownCloudVersion
+            parser.serverBaseUri = client.baseUri
+            result = parser.parse(putMethod.responseBodyAsString)
+
+            if (result.isSuccess && retrieveShareDetails) {
+                // retrieve more info - PUT only returns the index of the new share
+                val emptyShare = result.data.shares[0]
```
Comment (jcmontesmartos) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt]: Use first function better
`val emptyShare = result.data.shares.first()`

BTW, I am assuming you are checking shares is not empty, aren't you?

Comment (davigonz) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt]: Not really, the server returns the updated share with almost no information so I need to retrieve the share with all the information using another operation called GetRemoteShareOperation

File: build.gradle (hunk 1 above)
Comment (jcmontesmartos) [reply build.gradle]: You are brave enough to be always in the latest version of Android Studio but you don't want to test @Parcelize, interesting... 🤔

**Pull request merged**